document/js-api/index.bs

```diff
@@ -623,7 +634,21 @@ Immediately after a WebAssembly [=memory.grow=] instruction executes, perform th
 </div>
 
 <div algorithm>
-    The getter of the <dfn attribute for="Memory">buffer</dfn> attribute of {{Memory}} returns the receiver's \[[BufferObject]] internal slot.
+    The getter of the <dfn attribute for="Memory">buffer</dfn> attribute of {{Memory}} perfoms the following steps:
+
+    1. Let |memory| be the Memory instance.
+    1. Let |memaddr| be |memory|.\[[Memory]].
+    1. Let |block| be a [=Data Block=] which is [=identified with=] the underlying memory of |memaddr|.
+    1. If |block| is a [=Data Block=],
```

Comment:
This is kinda strange; a "Shared Data Block" has to be and not be a "Data Block" in these two steps.

Reply:
I would prefer relying on the shared attibute.

Reply:
It's weird, but I couldn't really see another way to distinguish a data block [in the ECMAScript spec](https://tc39.github.io/ecma262/#sec-data-blocks):

> A data block that resides in memory that can be referenced from multiple agents concurrently is designated a Shared Data Block. ... Finally, Shared Data Blocks can be distinguished from Data Blocks.

> I would prefer relying on the shared attibute.

That feels a bit more indirect to me, but seems like it would work too.

Reply:
@xtuc "shared" is just part of the dictionary passed into the constructor, and it's not present at this point.

Another way to phrase things would be to reverse the two conditionals, so it's "If block is a Shared Data Block/otherwise". That might be more clear.

Reply:
> "shared" is just part of the dictionary passed into the constructor, and it's not present at this point.

Oh, I assumed @xtuc meant the shared attribute of the wasm-side memory instance.

Reply:
> Another way to phrase things would be to reverse the two conditionals, so it's "If block is a Shared Data Block/otherwise". That might be more clear.

Done.